templates/documents/general_view.html

```diff
@@ -5,11 +5,21 @@
  * @link      http://www.open-emr.org
  * @author    Brady Miller <brady.g.miller@gmail.com>
  * @copyright Copyright (c) 2017-2019 Brady Miller <brady.g.miller@gmail.com>
+ * @author    Jerry Padgett <sjpadgett@gmail.com>
+ * @copyright Copyright (c) 2018-2019 Jerry Padgett <sjpadgett@gmail.com>
  * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  *}
+{php}
+use OpenEMR\Events\PatientDocuments\PatientDocumentEvent;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
 
+$eventDispatcher = $GLOBALS['kernel']->getEventDispatcher();
+{/php}
```

Comment:
We don't allow php tags in smarty anymore since these were all removed to support new version of smarty. We'll need to look into how to support this without breaking out of smarty.

Reply:
Actually, looks like we just need a quick smarty function to support this stuff. Let me know if you want me to do this. I'd be happy to oblige :)

Reply:
guess we'll have to. this is why I don't like template engines. always a compromise! I can't think of a way to do in controller at the moment...

Reply:
come to think of it, I don't like anything anymore. Can't wait 'till seventy.:)

Reply:
yep, figuring out smarty and templating took me way longer than other parts of OpenEMR; templating causes indigestion :)

Reply:
Will need to add this script at library/smarty/plugins/function.dispatchPatientDocumentEvent.php :
```
/**
 * Smarty plugin
 * @package Smarty
 * @subpackage plugins
 * dispatchPatientDocumentEvent() version for smarty templates
 *
 * Copyright (C) 2019 Brady Miller <brady.g.miller@gmail.com>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 */

use OpenEMR\Events\PatientDocuments\PatientDocumentEvent;
use Symfony\Component\EventDispatcher\EventDispatcherInterface;
use Symfony\Component\EventDispatcher\GenericEvent;

/**
 * Smarty {dispatchPatientDocumentEvent} function plugin
 *
 * Type:     function<br>
 * Name:     dispatchPatientDocumentEvent<br>
 * Purpose:  Event listener for PatientDocumentEvent<br>
 *
 * Examples:
 *
 * {dispatchPatientDocumentEvent event="javascript_ready_fax_dialog"}
 * {dispatchPatientDocumentEvent event="actions_render_fax_anchor"}
 *
 * @param array
 * @param Smarty
 */


function smarty_function_xlt($params, &$smarty)
{
    if (empty($params['event'])) {
        $smarty->trigger_error("dispatchPatientDocumentEvent: missing 'event' parameter");
        return;
    } else {
        $event = $params['event'];
    }

    if ($event == "javascript_ready_fax_dialog")  {
        $eventDispatcher->dispatch(PatientDocumentEvent::JAVASCRIPT_READY_FAX_DIALOG, new GenericEvent());
    } else if ($event == "actions_render_fax_anchor") {
        $eventDispatcher->dispatch(PatientDocumentEvent::ACTIONS_RENDER_FAX_ANCHOR, new GenericEvent());
    } else {
        $smarty->trigger_error("dispatchPatientDocumentEvent: invalid 'event' parameter");
        return;
    }
}
```

Reply:
lol I just pushed old view smarty. no problem, might have to use on 502. Will try your solution so bare with me as i'm scatter brain lately. too many drugs. 
You're a good man Dr. Miller!

Reply:
@bradymiller Works like a champ. Just have an escaping issue with file path. too many /\ mixed separators. js_escape breaks but js_attr works fine. need to have a file path escape function that normalizes paths to linux schema. I have one somewhere i'll dig out.
Anyway thanks Brady, a little work should be able to make the plugin generic for other events.

will put up changes in a few...